src/main/java/duke/Storage.java

```
@@ -0,0 +1,102 @@
+
+                    if (parts.length < 3) continue;
+
+                    String kind = parts[0];
+                    boolean done = "1".equals(parts[1]);
+                    String desc = parts[2];
+                    Task t;
+                    switch (kind) {
+                        case "T":
+                            t = new Todo(desc);
+                            break;
+                        case "D":
+                            if (parts.length < 4) continue;
+                            Task d;
+                            try {
+                                d = new Deadline(desc, java.time.LocalDate.parse(parts[3]));
+                            } catch (Exception ex) {
+                                d = new Deadline(desc, parts[3]);
+                            }
+                            t = d;
+                            break;
+                        case "E":
+                            if (parts.length < 5) continue;
+                            t = new Event(desc, parts[3], parts[4]);
+                            break;
+                        default:
+                            continue;
+                    }
+                    if (done) t.mark();
+                    tasks.add(t);
+                }
+            }
+            return tasks;
+        } catch (IOException e) {
+            return new TaskList();
+        }
+    }
+
+    public void save(TaskList tasks) {
+        try {
+            if (filePath.getParent() != null) {
+                Files.createDirectories(filePath.getParent());
+            }
+            try (BufferedWriter bw = Files.newBufferedWriter(filePath, StandardCharsets.UTF_8)) {
+                for (int i = 0; i < tasks.size(); i++) {
+                    Task t = tasks.get(i + 1);
+                    bw.write(serialize(t));
+                    bw.newLine();
+                }
+            }
+        } catch (IOException e) {
+
+        }
```

Maybe consider not having an empty catch statement?

```suggestion
        } catch (IOException e) {

        }
```